inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -18,79 +18,288 @@
         try:
-            conn = sqlite3.connect(self._db_file)
-            if isinstance(task, SingleSchedule):
-                self._create_single_schedule_task(task, conn)
-            elif isinstance(task, RepeatedSchedule):
-                self._create_repeated_schedule_task(task, conn)
+            if isinstance(schedule, SingleSchedule):
+                self._create_single_schedule(schedule)
+            elif isinstance(schedule, RepeatedSchedule):
+                self._create_repeated_schedule(schedule)
             else:
-                print("Task is neither a SingleSchedule nor a RepeatedSchedule object.")
-        except (sqlite3.OperationalError, sqlite3.InternalError) as e:
-            raise DispatcherException(f"Error connecting to database: {e}")
-        finally:
-            if conn:
-                conn.close()
+                logger.error("Schedule type is neither a SingleSchedule nor a RepeatedSchedule object.")
+        except (sqlite3.Error) as e:
+            raise DispatcherException(f"Error connecting to Dispatcher Schedule database: {e}")
+    
+    def _create_single_schedule(self, ss: SingleSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        logger.debug(f"Execute -> INSERT INTO single_schedule_manifest(request_id, start_time, end_time) VALUES({ss.request_id}{ss.start_time}{ss.end_time})")
 
-    def _create_single_schedule_task(self, task: SingleSchedule, conn: sqlite3.Connection) -> None:
-        # Creating the table
-        cur = conn.cursor()
-        # Check if the table already exists, if not, create the table
-        logger.debug("Checking if the table exist...")
-        cur.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='single_task'")
-        if cur.fetchone() is None:
-            logger.debug("Table not exist. Creating the table.")
-            sql = ''' CREATE TABLE single_task (
-                                    request_id text,
-                                    start_time text,
-                                    end_time text,
-                                    manifests text
-                                ); '''
-            conn.execute(sql)  # Execute the SQL command
+        sql = ''' INSERT INTO single_schedule(request_id, start_time, end_time)
+                                VALUES(?,?,?); '''
+        start_time = None if not ss.start_time else str(ss.start_time)
+        end_time = None if not ss.end_time else str(ss.end_time)
+        try:
+            self._conn.execute('BEGIN')
+            self._cursor.execute(sql, (ss.request_id, start_time, end_time))
+            schedule_id = self._cursor.lastrowid
+            if not schedule_id:
+                raise DispatcherException("No schedule id was added to the single_schedule table.")
+            
+            logger.debug(f"Added schedule with id: {str(schedule_id)}, request_id: {ss.request_id}, start_time: {str(ss.start_time)}, end_time: {str(ss.end_time)}")
```

Comment:
debug statement should match what's passed in to execute: start_time and end_time rather than str(ss.start_time) and str(ss.end_time)

Reply:
fixed.